From d7e32e36159c708d7b8be2ce91798fdd08fee0df Mon Sep 17 00:00:00 2001
From: Dave Gerlach <d-gerlach@ti.com>
Date: Tue, 13 Dec 2016 16:06:24 -0600
Subject: [PATCH 047/205] soc: ti: pm33xx: Add printing of wakeup source after
 suspend transition

This commit comes from:
  git://git.ti.com/processor-sdk/processor-sdk-linux.git

The wkup_m3 provides the wakeup source back to the system so use the
wkup_m3_ipc driver to get the wakeup source for the system after
a suspend transition.

Signed-off-by: Dave Gerlach <d-gerlach@ti.com>
(cherry picked from commit e230e229e956ac61d74d30a2fc4345ff7441df79)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/soc/ti/pm33xx.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/soc/ti/pm33xx.c b/drivers/soc/ti/pm33xx.c
index f567a9e..44600bd 100644
--- a/drivers/soc/ti/pm33xx.c
+++ b/drivers/soc/ti/pm33xx.c
@@ -90,6 +90,9 @@ static int am33xx_pm_suspend(suspend_state_t suspend_state)
 				"PM: CM3 returned unknown result = %d\n", i);
 			ret = -1;
 		}
+
+		pr_info("PM: Wakeup source %s\n",
+			m3_ipc->ops->request_wake_src(m3_ipc));
 	}
 
 	return ret;
-- 
1.7.9.5

